### ISSUE TITLE
Circe is not usable as a drop in replacement for UPickle

### ISSUE THREAD

**Issue by threeseed** [shogowada/scala-json-rpc#43]
Related to #41.

If you take the existing E2E examples and simply swap out UPickle with Circe you will get the following compiler errors:

````
could not find implicit value for parameter encoder: io.circe.Encoder[io.github.shogowada.scala.jsonrpc.Models.JSONRPCRequest[(String,)]]
[error]       <((new TodoListView(serverAndClient.createAPI[TodoRepositoryAPI])) ()).empty,
````
````
could not find implicit value for parameter decoder: io.circe.Decoder[io.github.shogowada.scala.jsonrpc.Models.JSONRPCId]
[error]         serverAndClient.receiveAndSend(message).onComplete {
````

There should be documentation (and ideally the code) that makes it clear to users that they need to implement the encoder/decoders for the internal Scala JSON RPC classes e.g. JsonRPCId. Both the semi-auto and auto don't appear to work in this case.

**Comment by shogowada** [comment]
Thanks for reporting. circe-json-serializer should include serializer for `JSONRPCId` out of the box. I will take a look this week.

**Comment by shogowada** [comment]
@baank It is now released as v0.9.3.
